### PR TITLE
chore: Add correct type casts to eip7594 functions

### PR DESCRIPTION
**Pull request by kevaundray**
cc @hwwhww 

This PR resolves all of the errors for eip7594.

I applied a workarounf for the error `error: Invalid base class "List"` -- Happy to revert if there is a better solution.

### CODE REVIEW

File: specs/_features/eip7594/polynomial-commitments-sampling.md [hunk 1]
```diff
@@ -484,6 +485,7 @@ def verify_cell_proof_batch(row_commitments_bytes: Sequence[Bytes48],
 
 ```python
 def construct_vanishing_polynomial(missing_cell_ids: Sequence[CellID]) -> Tuple[
+        Sequence[BLSFieldElement],
```
Comment (kevaundray) [inline specs/_features/eip7594/polynomial-commitments-sampling.md:488]: The third parameter was not being used, which is likely why this did not trigger any runtime errors.

I can remove it in a subsequent PR (Did not want to make any changes that were not geared towards satisfying pylint)

**Comment by kevaundray** [comment]
The invalid base class error comes from `ExtendedMatrix` | `List[Cell, MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS]`

I suspect that the issue arises because the size of the list is not a literal

File: specs/_features/eip7594/das-core.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ We define the following Python custom types for type hinting and readability:
 | Name | SSZ equivalent | Description |
 | - | - | - |
 | `DataColumn` | `List[Cell, MAX_BLOB_COMMITMENTS_PER_BLOCK]` | The data of each column in EIP-7594 |
-| `ExtendedMatrix` | `List[Cell, MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS]` | The full data of one-dimensional erasure coding extended blobs (in row major format) |
+| `ExtendedMatrix` | `List[Cell]` | The full data of one-dimensional erasure coding extended blobs (in row major format). It's max size is `MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS`|
```
Comment (kevaundray) [inline specs/_features/eip7594/das-core.md:49]: This is one possible workaround for the invalid base class issue

File: specs/_features/eip7594/das-core.md [hunk 3]
```diff
@@ -55,6 +55,7 @@ We define the following Python custom types for type hinting and readability:
 | Name | Value | Description |
 | - | - | - |
 | `NUMBER_OF_COLUMNS` | `uint64(FIELD_ELEMENTS_PER_EXT_BLOB // FIELD_ELEMENTS_PER_CELL)` (= 128) | Number of columns in the extended data matrix. |
+| `MAX_EXTENDED_MATRIX_SIZE` | `uint64(MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS)` (= 768) | The data size of `ExtendedMatrix`. |
```
Comment (hwwhww) [inline specs/_features/eip7594/das-core.md:58]: @kevaundray 

Re: the `error: Invalid base class "List"` issue

Pylint can't interpret `class ContainerName(List[c, a*b])` correctly. 

I added a workaround by adding a new config, `MAX_EXTENDED_MATRIX_SIZE`, to replace `AX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS`.

Another workaround is to treat it as a special case by adding a `# type: ignore` by updating
https://github.com/ethereum/consensus-specs/blob/9c04cf13c95a355b95b09d605fd10185b0995b41/pysetup/helpers.py#L25-L29

I prefer the former a bit.

File: specs/_features/eip7594/polynomial-commitments-sampling.md [hunk 4]
```diff
@@ -278,9 +279,9 @@ def vanishing_polynomialcoeff(xs: Sequence[BLSFieldElement]) -> PolynomialCoeff:
     """
     Compute the vanishing polynomial on ``xs`` (in coefficient form)
     """
-    p = [1]
+    p = PolynomialCoeff([1])
     for x in xs:
-        p = multiply_polynomialcoeff(p, [-int(x), 1])
+        p = multiply_polynomialcoeff(p, PolynomialCoeff([-int(x), 1]))
```
Comment (hwwhww) [inline specs/_features/eip7594/polynomial-commitments-sampling.md:284]: After d5b4e6cc6fc66b60882f1fcb29892f2fcfc1ab3c fix, we now have a error here:
```
eth2spec/eip7594/minimal.py:5014: in vanishing_polynomialcoeff
    p = multiply_polynomialcoeff(p, PolynomialCoeff([-int(x), 1]))
/Users/hww/.pyenv/versions/pypy3.9-arm64/envs/pypy39-202209/lib/pypy3.9/site-packages/remerkleable/complex.py:289: in __new__
    input_views.append(elem_cls.coerce_view(el))
/Users/hww/.pyenv/versions/pypy3.9-arm64/envs/pypy39-202209/lib/pypy3.9/site-packages/remerkleable/basic.py:192: in coerce_view
    return cls(v)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'eth2spec.eip7594.minimal.BLSFieldElement'>, value = -52435875175126190...3658699938581184512

    def __new__(cls, value: int):
        if value < 0:
>           raise ValueError(f"unsigned type {cls} must not be negative")
E           ValueError: unsigned type <class 'eth2spec.eip7594.minimal.BLSFieldElement'> must not be negative

/Users/hww/.pyenv/versions/pypy3.9-arm64/envs/pypy39-202209/lib/pypy3.9/site-packages/remerkleable/basic.py:81: ValueError
```

File: specs/_features/eip7594/das-core.md (hunk 3 above)
Comment (kevaundray) [reply specs/_features/eip7594/das-core.md:58]: I have no preference, so whatever you think is best :)

File: specs/_features/eip7594/polynomial-commitments-sampling.md (hunk 4 above)
Comment (kevaundray) [reply specs/_features/eip7594/polynomial-commitments-sampling.md:284]: The fix is to do `BLS_MODULUS - int(x)` so the value is in the correct range -- I'm having trouble running `make test`, so I'm gonna push a blind fix for now

**Comment by ralexstokes** [comment]
is this PR still active? or can we close for being stale? looks like the problem this PR addresses has been handled in other ways since?

**Comment by kevaundray** [comment]
> is this PR still active? or can we close for being stale? looks like the problem this PR addresses has been handled in other ways since?

I've merged in most of the changes for this in separate PRs, we haven't enabled type casts because the CI kept timing out (happy to close and re-open a new PR when I have more time to investigate the typing issue -- the code has been updated since)

**Comment by Aga92-92** [comment]
.

**Comment by Aga92-92** [comment]
.

**Pull request closed without merging**